libcontainer/container_linux.go

```diff
@@ -201,18 +201,30 @@ func (c *Container) Set(config configs.Config) error {
 
 // Start starts a process inside the container. Returns error if process fails
 // to start. You can track process lifecycle with passed Process structure.
-func (c *Container) Start(process *Process) error {
+func (c *Container) Start(process *Process) (retErr error) {
 	c.m.Lock()
 	defer c.m.Unlock()
+	defer func() {
+		if retErr != nil {
+			c.terminate(process)
+		}
+	}()
```

Comment:
To me, this looks overcomplicated, given that this function only calls c.start.

An alternative would be

```go
func (c *Container) Start(process *Process) error {
	c.m.Lock()
	defer c.m.Unlock()
	if err := c.start(process); err != nil {
		c.terminate(process)
		return err
	}
}
```